plugins/com.google.cloud.tools.eclipse.appengine.facets/src/com/google/cloud/tools/eclipse/appengine/facets/AppEngineStandardFacet.java

```diff
@@ -257,4 +269,51 @@ private static void addWebFacetToBatch(IFacetedProject facetedProject,
     return appEngineRuntimes.toArray(appEngineRuntimesArray);
   }
 
+  /**
+   * Attempts to find a main web application directory, by the following logic:
+   *
+   * 1. If there is no {@code WEB-INF} folder in the {@code project}, returns {@code null}.
+   * 2. Otherwise, if there is at least one {@code WEB-INF} folder that contains {@code web.xml},
+   *     returns the parent directory of one of such {@code WEB-INF} folders.
+   * 3. Otherwise, returns the parent directory of an arbitrary {@code WEB-INF}.
```

Comment:
should we also consider the presence/absence of appengine-web.xml when choosing between multiple candidates?